libraries/rib-base/src/main/java/com/badoo/ribs/routing/source/impl/Pool.kt

```diff
@@ -1,36 +1,80 @@
 package com.badoo.ribs.routing.source.impl
 
+import android.os.Bundle
 import android.os.Parcelable
+import com.badoo.ribs.core.modality.BuildParams
 import com.badoo.ribs.minimal.reactive.Cancellable
 import com.badoo.ribs.minimal.reactive.Relay
+import com.badoo.ribs.minimal.state.TimeCapsule
 import com.badoo.ribs.routing.Routing
 import com.badoo.ribs.routing.history.RoutingHistory
 import com.badoo.ribs.routing.history.RoutingHistoryElement
+import com.badoo.ribs.routing.history.RoutingHistoryElement.Activation
 import com.badoo.ribs.routing.source.RoutingSource
+import kotlinx.parcelize.Parcelize
 import java.util.UUID
 
 class Pool<C : Parcelable>(
-    private val allowRepeatingConfigurations: Boolean
+    initialItems: List<Item<C>>,
+    private val allowRepeatingConfigurations: Boolean,
+    private val timeCapsule: TimeCapsule
 ) : RoutingSource<C> {
-    private var elements: Map<Routing.Identifier, RoutingHistoryElement<C>> = emptyMap()
+
+    @Parcelize
+    data class State<C : Parcelable>(
+        val elements: Map<Routing.Identifier, RoutingHistoryElement<C>> = emptyMap()
+    ) : Parcelable
+
+    private var elements: Map<Routing.Identifier, RoutingHistoryElement<C>> = HashMap(timeCapsule.elements())
     private val current: RoutingHistory<C>
         get() = RoutingHistory.from(
-            elements.values
+            elements.values.toList()
         )
 
     private val states: Relay<RoutingHistory<C>> = Relay()
 
-    // FIXME save/restore to bundle to support correct baseLineState
+    init {
+        timeCapsule.register(TIME_CAPSULE_KEY) { State(elements) }
+        if (elements.isEmpty()) {
```

Comment:
We can't distinct 2 cases here:
1. We created Pool without bundle available
2. We removed everything from Pool manually and restore it from bundle

In the second case we will reinitialize initial items instead of keeping Pool empty.